Simplify the bookinfo instructions for auto injection (#3393)
istio/istio.io: content/docs/examples/bookinfo/index.md

If you use GKE, please ensure your cluster has at least 4 standard GKE nodes. If you use Minikube, please ensure you have at least 4GB RAM.
{{< /tip >}}

> 1.  Change directory to the root of the Istio installation directory.

Reviewer: ```suggestion
1.  Change directory to the root of the Istio installation.
```

Reviewer 2: ```suggestion
1. Go to the root directory of your Istio installation.
```

1.  The default Istio installation uses [automatic sidecar injection](/docs/setup/kubernetes/sidecar-injection/#automatic-sidecar-injection).

>     Label the namespace that will host the application with `istio-injection=enabled`:

Reviewer 2: How are the previous statement and this one connected? Do readers need to know the default behavior at this time to complete the auto injection? If the sidecar is injected automatically by default, why do I need to add the label to the `default` namespace?

Author: It uses automatic injection but it's on a per-namespace, so the namespace still has to be labeled. Does it make sense?

    {{< text bash >}}
    $ kubectl label namespace default istio-injection=enabled
    {{< /text >}}

> 1.  Deploy your application using `kubectl` as usual:

Reviewer 2: Avoid using "as usual". We don't know what is usual to our readers. Opt instead for more direct explicit language:
```suggestion
1.  To deploy your application, use this `kubectl` command:
```

Author: I meant to emphasize the magic the auto-injection does.

Author: Removed "as usual"

    {{< text bash >}}
    $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo.yaml@
    {{< /text >}}

    The above command launches all four microservices as illustrated in the above diagram.
    All 3 versions of the reviews service, v1, v2, and v3, are started.

    {{< tip >}}
    In a realistic deployment, new versions of a microservice are deployed
    over time instead of deploying all versions simultaneously.
    {{< /tip >}}

>     *Note: If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]

Reviewer 3: Please use {{<tip>}} or {{<warning>}} instead of *Note

>     *Note: If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]

Reviewer: ```suggestion
    {{ <tip> }}
    If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]
```

Reviewer 2: Do not overuse callouts. There is no reason for this to be a note, tip or similar. Make it a simple paragraph adding context. A simple sentence providing the link is sufficient.
```suggestion
    If you disabled automatic sidecar injection during installation, visit our [manual sidecar injection guide](/docs/setup/kubernetes/sidecar-injection/#manual-sidecar-injection) for more information.
```

    (/docs/setup/kubernetes/sidecar-injection/#manual-sidecar-injection), use the `istioctl kube-inject` command
     to modify the `bookinfo.yaml` file before deploying your apoplication,

>      as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject):*

Reviewer: ```suggestion
     as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject).
     {{ </tip> }}
```

    {{< text bash >}}
    $ kubectl apply -f <(istioctl kube-inject -f @samples/bookinfo/platform/kube/bookinfo.yaml@)

>     {{< /text >}}

Reviewer: Oh, if you want to include the command, then I guess you can't use a tip, as suggested above. Maybe best, is to just make the previous part an ordinary paragraph, i.e., no Note:
```
If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection],
as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject), use the following command
instead:
```

Reviewer 2: See my previous comment. Do not include the command. We cannot provide the necessary context for it to be useful to someone reading this doc. We have an entire document covering manual sidecar injection and we should try to keep this flow simple.

Author: Are you both saying this section should be removed?

Reviewer: I would keep the tip, but I think you need to delete the `text bash` command, because you can't nest it inside of the tip.

I would suggest changing it to `{{ warning }}`, instead of tip, and move it up just after the command, but before the text "The above command ...".
```
    {{< warning >}}
    If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]
    (/docs/setup/kubernetes/sidecar-injection/#manual-sidecar-injection), use the `istioctl kube-inject` command
    to modify the `bookinfo.yaml` file before deploying your apoplication,
    as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject).
    {{< /warning >}}

    The above command launches all four microservices as illustrated in the above diagram.
    All 3 versions of the reviews service, v1, v2, and v3, are started.
```

Author: It does not work indeed, I moved it outside of the tip.
This is still a tip, not a warning.

Author: I checked and it works, but the linter complains, so it does not really work for now.

If you use GKE, please ensure your cluster has at least 4 standard GKE nodes. If you use Minikube, please ensure you have at least 4GB RAM.
{{< /tip >}}

1.  Change directory to the root of the Istio installation.

1.  The default Istio installation uses [automatic sidecar injection](/docs/setup/kubernetes/additional-setup/sidecar-injection/#automatic-sidecar-injection).
    Label the namespace that will host the application with `istio-injection=enabled`:

    {{< text bash >}}
    $ kubectl label namespace default istio-injection=enabled
    {{< /text >}}

1.  Deploy your application using the `kubectl` command:

    {{< text bash >}}
    $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo.yaml@
    {{< /text >}}

    The command launches all four services shown in the `bookinfo` application architecture diagram.
    All 3 versions of the reviews service, v1, v2, and v3, are started.

    {{< tip >}}
    In a realistic deployment, new versions of a microservice are deployed
    over time instead of deploying all versions simultaneously.
    {{< /tip >}}

    If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]

>     (/docs/setup/kubernetes/additional-setup/sidecar-injection/#manual-sidecar-injection), 

Reviewer 2: ```suggestion
    (/docs/setup/kubernetes/additional-setup/sidecar-injection/#manual-sidecar-injection),
```

If you use GKE, please ensure your cluster has at least 4 standard GKE nodes. If you use Minikube, please ensure you have at least 4GB RAM.
{{< /tip >}}

> 1.  Change directory to the root of the Istio installation.

Reviewer 2: ```suggestion
1.  Go to the root directory of the Istio installation.
```

Author: Nack, change directory is the correct operation.

1.  The default Istio installation uses [automatic sidecar injection](/docs/setup/kubernetes/additional-setup/sidecar-injection/#automatic-sidecar-injection).

>     Label the namespace that will host the application with `istio-injection=enabled`:

Reviewer 2: ```suggestion
    Use the `istio-injection=enabled` label for the namespace hosting the application:
```

Author: Nack, label is the correct operation.

    {{< text bash >}}
    $ kubectl label namespace default istio-injection=enabled
    {{< /text >}}

1.  Deploy your application using the `kubectl` command:

    {{< text bash >}}
    $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo.yaml@
    {{< /text >}}

>     The above command launches all four microservices as illustrated in the above diagram.

Reviewer 2: Do not reference images that are not immediately above or below the reference. Doing so, confuses readers.
```suggestion
    The command launches all four services shown in the `bookinfo` application architecture diagram.
```

    All 3 versions of the reviews service, v1, v2, and v3, are started.

    {{< tip >}}
    In a realistic deployment, new versions of a microservice are deployed
    over time instead of deploying all versions simultaneously.
    {{< /tip >}}

>     {{< tip >}}

Reviewer 3: Please use tips, warnings, and ideas sparingly. They are meant for high impact things not for normal paragraphs.

Reviewer 2: Absolutely agree with @geeknoid By definition tips are used to convey brief optional information to readers and should be used sparingly.

Author: That's a matter of taste.

Author: Removed nevertheless.

    If you disabled automatic sidecar injection during installation and rely on [manual sidecar injection]

>     (/docs/setup/kubernetes/additional-setup/sidecar-injection/#manual-sidecar-injection), use the `istioctl kube-inject` command

Reviewer 3: Could use a link to the istioctl command in the reference docs.

Author: Original code did not have it either, and adding links it's an issue when pages get re-organized.

    to modify the `bookinfo.yaml` file before deploying your apoplication,

>     as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject):

Reviewer 2: Use descriptive text for links. See: https://preliminary.istio.io/about/contribute/style-guide/#create-useful-links for more info.
```suggestion
    visit the `istioctl` [reference documentation](/docs/reference/commands/istioctl/#istioctl-kube-inject) for more information.
```

    {{< /tip >}}

    {{< text bash >}}

> 

Reviewer 3: Remove empty line here and at the end of the text block.

Author: Linter fails if I do this.
